packages/terra-list/src/ListItem.jsx

```diff
@@ -22,57 +26,73 @@ const propTypes = {
    */
   isSelectable: PropTypes.bool,
   /**
-   * Whether or not the list item has a disclosure indicator presented.
+   * The associated metaData to be provided in the onSelect callback.
    */
-  hasChevron: PropTypes.bool,
+  // eslint-disable-next-line react/forbid-prop-types
+  metaData: PropTypes.object,
+  /**
+   * Function callback for when the appropriate click or key action is performed.
+   * Callback contains the javascript evnt and prop metadata, e.g. onSelect(event, metaData)
+   */
+  onSelect: PropTypes.func,
   /**
    * Function callback for the ref of the li.
    */
   refCallback: PropTypes.func,
 };
 
 const defaultProps = {
-  content: undefined,
+  children: [],
+  hasChevron: false,
   isSelected: false,
-  isSelectable: undefined,
-  hasChevron: undefined,
+  isSelectable: false,
 };
 
 const ListItem = ({
-  content,
+  children,
+  hasChevron,
   isSelected,
   isSelectable,
-  hasChevron,
+  metaData,
+  onBlur,
+  onClick,
+  onKeyDown,
+  onMouseDown,
+  onSelect,
   refCallback,
   ...customProps
 }) => {
   const listItemClassNames = cx([
     'list-item',
     { selected: isSelected },
     { 'is-selectable': isSelectable },
+    { 'item-has-chevron': hasChevron },
     customProps.className,
   ]);
 
-  const ariaSpread = isSelectable ? { role: 'option', 'aria-selected': isSelected } : {};
+  const ariaSpread = {};
```

Comment:
nit: rename this to `attrSpread` as more than just aria attributes are being spread.

Reply:
Updated.